scipy/stats/stats.py

```diff
@@ -3708,10 +3708,10 @@ def weightedtau(x, y, rank=True, weigher=None, additive=True):
         return WeightedTauResult(np.nan, np.nan)  # Return NaN if arrays are empty
 
     # If there are NaNs we apply _toint64()
-    if np.isnan(np.min(x)):
-            x = _toint64(x)
-    if np.isnan(np.min(y)):
-            y = _toint64(y)
+    if np.isnan(x).any():
+        x = _toint64(x)
+    if np.isnan(y).any():
```

Comment:
I remember seeing a stackoverflow question where `np.isnan(np.ptp(y))` is faster

Reply:
Weird behaviour with np.min is what leads us to this point.